annotation-utils/src/main/java/eu/europeana/annotation/utils/parse/AnnotationLdParser.java

```diff
@@ -609,6 +626,78 @@ private Body parseBody(JSONObject valueObject, MotivationTypes motivation) throw
 		return body;
 	}
 
+	/**
+	 * @param body
+	 * @param streetAddress
+	 * @throws JsonParseException
+	 */
+	private void setVcardStreetAddress(Body body, String streetAddress) throws JsonParseException {
+		if (body instanceof VcardAddressBody)
+			((VcardAddressBody) body).getAddress().setVcardStreetAddress(streetAddress);
+		else
+			throw new JsonParseException("Street address not supported for bodyType: " + body.getInternalType());
+	}
+
+	/**
+	 * @param body
+	 * @param postalCode
+	 * @throws JsonParseException
+	 */
+	private void setVcardPostalCode(Body body, String postalCode) throws JsonParseException {
+		if (body instanceof VcardAddressBody)
+			((VcardAddressBody) body).getAddress().setVcardPostalCode(postalCode);
+		else
+			throw new JsonParseException("Street address not supported for bodyType: " + body.getInternalType());
+	}
+
+	/**
+	 * @param body
+	 * @param postOfficeBox
+	 * @throws JsonParseException
+	 */
+	private void setVcardPostOfficeBox(Body body, String postOfficeBox) throws JsonParseException {
+		if (body instanceof VcardAddressBody)
+			((VcardAddressBody) body).getAddress().setVcardPostOfficeBox(postOfficeBox);
+		else
+			throw new JsonParseException("Post office box not supported for bodyType: " + body.getInternalType());
+	}
+
+	/**
+	 * @param body
+	 * @param locality
+	 * @throws JsonParseException
+	 */
+	private void setVcardLocality(Body body, String locality) throws JsonParseException {
+		if (body instanceof VcardAddressBody)
+			((VcardAddressBody) body).getAddress().setVcardLocality(locality);
+		else
+			throw new JsonParseException("Locality not supported for bodyType: " + body.getInternalType());
+	}
+
+	/**
+	 * @param body
+	 * @param region
+	 * @throws JsonParseException
+	 */
+	private void setVcardHasGeo(Body body, String region) throws JsonParseException {
+		if (body instanceof VcardAddressBody)
+			((VcardAddressBody) body).getAddress().setVcardHasGeo(region);
+		else
+			throw new JsonParseException("Region not supported for bodyType: " + body.getInternalType());
+	}
+
+	/**
+	 * @param body
+	 * @param countryName
+	 * @throws JsonParseException
+	 */
+	private void setVcardCountryName(Body body, String countryName) throws JsonParseException {
```

Comment:
The first parameter must be of type VcardAddressBody the instanceOf checking and the exception throwing can be removed from the method

Reply:
removed instanceOf check and throwing exception